### PR TITLE
[2.3] Release-note updates from release review

### PR DESCRIPTION
**Pull request by glennmatthews**
Addressing some review feedback from #6035 regarding the release-note and `mkdocs.yml`.

### CODE REVIEW

**Review by HanlinMiao — APPROVED** [review]

File: nautobot/docs/release-notes/version-2.3.md [hunk 1]
```diff
@@ -8,37 +8,53 @@ This document describes all new features and changes in Nautobot 2.3.
...
 
-#### Dynamic Group Cache Changes
+#### Changed TreeManager Default Behavior ([#5786](https://github.com/nautobot/nautobot/issues/5786))
+
+The `TreeManager` class (used for tree-models such as Location, RackGroup, and TenantGroup) default behavior has changed from `with_tree_fields` to `without_tree_fields`. This should improve performance in many cases but may impact Apps or Jobs that were relying on the old default; such code should be updated to explicitly call `.with_tree_fields()` where appropriate.
+
+#### Consolidated Detail View Action Buttons with a Dropdown Menu ([#5871](https://github.com/nautobot/nautobot/issues/5871))
+
+Consolidate common standard action buttons (those at the top right) in all detail views with a bootstrap split button dropdown.
+
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:56]: If we're going to call this one out, we should probably also call out #5870 and #5869 as well, probably in a single section titled something like "UI Button Consolidation"

Comment (glennmatthews) [reply nautobot/docs/release-notes/version-2.3.md:56]:
```suggestion
#### Consolidated UI Buttons ([#5869](https://github.com/nautobot/nautobot/issues/5869), [#5870](https://github.com/nautobot/nautobot/issues/5870), [#5871](https://github.com/nautobot/nautobot/issues/5871))

Various button groups in the "object list" and "object detail" views have been consolidated following a common UI pattern of a single button for the most common action plus a popup menu for less common actions.
```

Comment (HanlinMiao) [reply nautobot/docs/release-notes/version-2.3.md]: Oops sorry just saw this and pushed up a new commit. Lmk if the new change is acceptable?

Comment (glennmatthews) [reply nautobot/docs/release-notes/version-2.3.md]: I don't think users care that it's a `bootstrap split button dropdown` specifically, my reword above was an attempt to make it a little less jargony. LMK.

File: nautobot/docs/release-notes/version-2.3.md [hunk 2]
```diff
@@ -64,6 +72,16 @@ Cleanup of the change log (deletion of `ObjectChange` records older than a given
 
 As an additional enhancement, the `LogsCleanup` Job can also be used to cleanup `JobResult` records if desired as well.
 
+#### UI Button Consolidation ([#5869](https://github.com/nautobot/nautobot/issues/5869), [#5870](https://github.com/nautobot/nautobot/issues/5870), [#5871](https://github.com/nautobot/nautobot/issues/5871))
+
+Consolidate list view bulk action buttons, list view standard action buttons and detail view standard action buttons with bootstrap split button dropdowns.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:77]:
```suggestion
Various button groups in the "object list" and "object detail" views have been consolidated following a common UI pattern of a single button for the most common action plus a popup menu for less common actions.
```

**Pull request merged**